spec/spec.md

```diff
@@ -1013,62 +1012,197 @@ in activities performed by entities participating in the message thread.
-  "descriptor": { // Message Descriptor
-    "method": "PermissionsQuery",
-    "schema": "https://schema.org/MusicPlaylist",
+{
+  "descriptor": {
+    "method": "PermissionsRequest",
+    "objectId": "b6464162-84af-4aab-aff5-f1f8438dfc1e",
+    "requester": "did:example:bob",
+    "description": "Help you create and edit your music playlists",
+    "ability": {
+      "can": {
+        "method": "CollectionsWrite",
+        "schema": "https://schema.org/MusicPlaylist"
+      },
+      "conditions": {
+        "encryption": 1,
+        "attestation": 0,
+        "delegation": true,
+        "sharedAccess": true
+      }
+    }
+  },
+  "attestation": {
+    "protected": {
+      "alg": "ES256K",
+      "kid": "did:example:bob#key-1"
+    },
+    "payload": CID(descriptor),
+    "signature": Sign(protected + payload)
   }
 }
 ```
 
 #### Grant
 
+`PermissionsGrant` messages are JSON objects that include general [Message Descriptor](#message-descriptors) properties and the following additional properties, which ****must**** be composed as follows:
+
+- The message object ****MUST**** contain a `descriptor` property, and its value ****MUST**** be a JSON object composed as follows:
+  - The object ****MUST**** contain a `method` property, and its value ****MUST**** be the string `PermissionsRequest`.
+  - The object ****MUST**** contain an `objectId` property, and its value ****MUST**** be an [[spec:rfc4122]] UUID Version 4 string representing the reply object.
+  - If the granted permission is in response to a `PermissionRequest`, the object ****MUST**** contain a `grantedFor` property, and its value ****MUST**** be the [[spec:rfc4122]] UUID Version 4 string of the `PermissionRequest` object the permission is being granted in relation to.
+  - The object ****MUST**** contain a `cid` property, and its value ****MUST**** be the stringified [Version 1 CID](https://docs.ipfs.io/concepts/content-addressing/#identifier-formats) of the [DAG PB](https://github.com/ipld/specs/blob/master/block-layer/codecs/dag-pb.md) encoded JSON Web Token for the granted permission, as defined in the [Capability Objects](#capability-objects) section below.
+  - The object ****MUST**** contain a `dataFormat` property, and its value ****MUST**** be the string `application/json`, as all granted permissions are represented as JSON Web Tokens generally adherent to the [UCAN](https://github.com/ucan-wg/spec#325-attenuations) capabilities construction.
+- The message object ****MUST**** contain an `attestation` property, which ****MUST**** be a JSON object as defined by the [Signed Data](#signed-data) 
+  section of this specification, with the requirement that the `kid` and `signature` ****MUST**** match the DID of the requesting party.
+- The message will contain a `data` payload, which is a JSON Web Token representation of the granted permission as defined in the [Capability Objects](#capability-objects) section below.
+
 ```json
-{ // Message
-  "data": {...},
```

Comment:
`data` needs to include an encryption key if the permission has an encryption requirement